lib/api/controllers/serverController.js

```diff
@@ -264,6 +219,70 @@ class ServerController extends BaseController {
         return Bluebird.resolve({serverInfo: response});
       });
   }
+
+  publicApi () {
+    const
+      kuzzleApi = this._buildApiDefinition(
+        this.kuzzle.funnel.controllers,
+        this.kuzzle.config.http.routes
+      ),
+      pluginsApi = this._buildApiDefinition(
+        this.kuzzle.funnel.pluginsControllers,
+        this.kuzzle.pluginsManager.routes,
+        '_plugin/'
+      ),
+      apiDefinition = Object.assign({}, kuzzleApi, pluginsApi);
+
+    return Bluebird.resolve(apiDefinition);
+  }
+
+  _buildApiDefinition (controllers, httpRoutes, urlPrefix = '') {
```

Comment:
(optional) I wonder how much it would cost to get rid of that http routes configuration file and, instead, make each controller expose a list of HTTP routes for each action in their list of actions. That would greatly simplify routes retrieval.

Reply:
Yes we could totally do this, like for the plugin routes definition! Also we could think about an unified API for both internal controllers and plugin controllers so Kuzzle can handle them the same way.

But I think this is outside the scope of this PR.

Reply:
I agree, but this would be a nice refactor to do :)